src/main/java/org/broadinstitute/consent/http/resources/DataRequestVoteResource.java

```diff
@@ -105,12 +107,17 @@ public Response updateFinalAccessVote(Vote rec,
             Vote vote = api.firstVoteUpdate(rec, id);
             Document access = accessRequestAPI.describeDataAccessRequestById(requestId);
             List<Integer> dataSets = DarUtil.getIntegerList(access, DarConstants.DATASET_ID);
-            if(access.containsKey(DarConstants.RESTRICTION)){
-                List<Vote> votes = vote.getType().equals(VoteType.FINAL.getValue()) ? api.describeVoteByTypeAndElectionId(VoteType.AGREEMENT.getValue(), vote.getElectionId()) :  api.describeVoteByTypeAndElectionId(VoteType.FINAL.getValue(), vote.getElectionId());
-                if(vote.getVote() != null && votes.get(0).getVote() != null){
-                    electionAPI.updateFinalAccessVoteDataRequestElection(rec.getElectionId());
-                }
-            }else {
+            // Find any final or agreement votes for this election that have a populated vote:
+            List<Vote> votes = voteService
+                  .findVotesByElectionIds(Collections.singletonList(rec.getElectionId()))
+                  .stream()
+                  .filter(v ->
+                      v.getType().equalsIgnoreCase(VoteType.FINAL.getValue()) ||
+                      v.getType().equalsIgnoreCase(VoteType.AGREEMENT.getValue()))
+                  .filter(v -> Objects.nonNull(v.getVote()))
+                  .collect(Collectors.toList());
+            // If we have any votes with a populated vote, we can now update the election.
+            if (!votes.isEmpty()) {
```

Comment:
Previously, this call was making a confusing ternary that didn't make clear the intention. The goal here was to find out if there were any existing FINAL or AGREEMENT votes that had been voted on (i.e., have a non-null vote value). Such a value would mean that a chairperson had recorded the final vote for a DAR election. If any such vote existed, then it was appropriate to update the election's final access vote value.